index.sh

```diff
@@ -7,12 +7,20 @@ echo "["
 ## now loop through meta files
 for i in "${arr[@]}"
 do
-    IFS='/' segments=($i)
     if [ "$FIRST_LINE" = true ] ; then
-        echo \{\"name\":\"${segments[2]}\",\"version\":\"${segments[3]}\"\}
+        echo "{"
         FIRST_LINE=false
     else
-        echo \,\{\"name\":\"${segments[2]}\",\"version\":\"${segments[3]}\"\}
+        echo ",{"
     fi
+    # 1. read meta.yaml to stio
+    # 2. filter lines with name,version,type
+    # 3. Replace ` :` with `":"`
+    # 4. Append `",` to the end of each line
+    # 5. Append `"` at the begining of each line
+    # 6. Remove all new lines
+    # 7. Remove last ','
+    cat $i|grep -e name -e version -e type |   sed 's/: /\":"/g'  | sed 's/$/\",/g' | sed 's/^/\"/g'  | tr -d '\n' |  sed 's/,$//g'
```

Comment:
Helpful comments :+1: 
I think it can be made even better by reformatting in the following way:
```bash
    cat $i | \                             # read meta.yaml to stio
       grep -e name -e version -e type | \ # filter lines with name,version,type
       sed 's/: /\":"/g'  | \              # Replace ` :` with `":"`
     ...
       sed 's/,$//g'                       # Remove last ','
```
WDYT?